packages/aws-client/S3.js

```diff
@@ -156,6 +156,70 @@ exports.s3CopyObject = improveStackTrace(
   }).promise()
 );
 
+/**
+ * Copies an arbitrarily sized object from one S3 location to another by using
+ * a multipart upload, if the object is large enough, also copying over the
+ * tags, if any.
+ *
+ * @param {Object} params - same params as AWS.S3.copyObject
+ * @returns {Promise} a promise of the upload request
+ * @see {@link https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#copyObject-property|AWS.S3.copyObject}
+ */
+exports.s3CopyUpload = improveStackTrace(async (params) => {
+  const [srcBucket, srcKey] = exports.getFileBucketAndKey(params.CopySource);
+  const srcStream = exports.getS3ObjectReadStream(srcBucket, srcKey);
+  const tagging = await exports.s3GetObjectTagging(srcBucket, srcKey);
+  const uploadParams = {
+    Tagging: exports.encodeTags(tagging.TagSet),
+    ...params,
+    Body: srcStream,
+  };
+
+  delete uploadParams.CopySource;
+
+  return exports.promiseS3Upload(uploadParams);
+});
+
+/**
+ * Returns a encoded URL Query parameter string of the form `Key=Value`, where
+ * `Key` and `Value` are the values of those properties of the specified tag.
+ *
+ * @param {Object} tag - the tag to be encoded
+ * @param {string} tag.Key - the tag's key
+ * @param {string} tag.Value - the tag's value
+ * @returns {string} encoded URL Query parameter string of the form `Key=Value`
+ */
+const encodeTag = ({ Key, Value }) =>
```

Comment:
Comment (not review change)  I think this might be a candidate for our jsdoc-optional discussion in the arch meeting